### PR TITLE
Fix problems with script contents after cloning

### PR DESCRIPTION
**Pull request by davidsbailey**
A number of problems exist today when cloning a script using `Script#clone_with_suffix`. This PR fixes the following problems which appear within the new `.script` file itself:
1. the script name should not contain the old suffix
2. `is_stable` should always be `false`
3. `script_announcements` should be cleared
4.  infer `version_year` from the new suffix if possible

### CODE REVIEW

**Comment by uponthesun** [comment]
[Jenkins] test this please

**Comment by davidsbailey** [comment]
@islemaster , feel free to ping me for a rundown on script copying prior to reviewing -- unfortunately Ram had been my go-to person for reviews in this area.

File: dashboard/test/fixtures/test-fixture-versioned-1801.script [hunk 1]
```diff
@@ -0,0 +1,8 @@
+family_name 'versioned'
+version_year '1800'
```
Comment (islemaster) [inline dashboard/test/fixtures/test-fixture-versioned-1801.script]: Does this intentionally not match the suffix on this fixture to prove that we're not depending on the previous `version_year` property?

**Review by islemaster — APPROVED** [review]
:+1: Clear tests made it easy to understand what's going on here, even without much past context.  Nice work!

**Comment by davidsbailey** [comment]
merging after confirming that (1) UI tests passed on a previous run, and  (2) only UI tests failed on subsequent runs after changes which should not affect UI tests.

File: dashboard/test/fixtures/test-fixture-versioned-1801.script (hunk 1 above)
Comment (davidsbailey) [reply dashboard/test/fixtures/test-fixture-versioned-1801.script]: oops, no, this was a typo. fixed now.

**Pull request merged**